default.nix

```diff
@@ -0,0 +1,122 @@
+
+# This builds a statically-linked `spago` executable.
+# It currently only works on Linux.
+
+let
+  nixpkgs-src = builtins.fetchTarball {
+    # This is one of the last commits from https://github.com/NixOS/nixpkgs/pull/169513.
+    # This commit is used because it is for the `haskell-updates` branch.
+    # Hydra builds a GHC for static linking on this branch, so we can get
+    # most dependencies from the NixOS cache.
+    url = "https://github.com/NixOS/nixpkgs/archive/999a0e87f9b1792e229c53602e80bda6dd52c105.tar.gz";
+    sha256 = "1p0vgmdz884gld4p2zcdy7g0w7xznc4kxj80wf5pk2qdxicbnscy";
+  };
```

Comment:
One problem with building from Nixpkgs like this is that we get the Haskell package versions in whatever commit of Nixpkgs we use.  So with the above commit, we are building with mainly LTS-19, while the `stack.yaml` in this repo is still LTS-18.

While this discrepancy doesn't seem to cause any big problems now, it is at least unfortunate that we're not using the same exact package versions.

([haskell.nix](https://github.com/input-output-hk/haskell.nix) has tools for generating a Haskell package set using versions from a `stack.yaml` and Stackage resolver, but haskell.nix comes with its own set of problems.)

Reply:
That's a good point, and a good reason for us to make sure we use the same package versions.

> Would it make sense to generate the package versions in the stack.yaml from the ones in Nixpkgs? (and of course check that they match in CI)

I think this would make sense, although I don't know of any sort of tool that is able to generate a `stack.yaml` from a given Nixpkgs commit.  I'm not sure how difficult it would be to write something like this.

For now, it would probably be the easiest to just manually keep the Nixpkgs Haskell package set and the resolver from the `stack.yaml` file the same.  This PR uses Nixpkgs at commit https://github.com/NixOS/nixpkgs/commit/999a0e87f9b1792e229c53602e80bda6dd52c105, which is included in PR https://github.com/NixOS/nixpkgs/pull/169513.  At the top of that PR, you can see there is a commit bumping the Nixpkgs Haskell stuff to LTS-19.4.

So maybe I could take a stab at bumping spago to LTS-19.4 (possibly in this PR, or a separate PR).  Then we'd be using the same LTS version and GHC version in both `stack.yaml` and the Nixpkgs Haskell package set.

> This would mean depending on whatever versions of packages coming from Nixpkgs, which is not the best (because we introduce a dependency) but it's pretty alright in the end.

I agree this is somewhat unfortunate.  Ideally there would be a tool that could generate a Nixpkgs Haskell package set from a `stack.yaml` file, but we don't currently have something like that.  (There used to be a tool https://github.com/input-output-hk/stack2nix that would do this, but it hasn't been maintained over the last few years.)

> I assume we would/could tolerate some overrides in case we need them for the build to work.

Yeah, the Nixpkgs Haskell side is somewhat flexible, and it should be possible to add any overrides that we may need.  I'd of course be happy to help out with this.